File: filter.js
function filter(inputStrings, predicateFn) {
    const result = [];
    inputStrings.forEach(function(item) {
        if (predicateFn(item)) {
            result.push(item);
        }
    });
    return result;
}

function checkStringStartsWithAny(startChar) {
    let predicate = (inputString) => inputString.charAt(0) === startChar;
    return predicate;
}

function printArrayToConsole(inputArray) {
    inputArray.forEach(function(item) {
        console.log(item);
    });
}

let strings = ["Hello", "World", "Hi", "Am", "a"];
let filteredStrings = filter(strings, checkStringStartsWithAny("A"));
printArrayToConsole(filteredStrings);

File: filter.py
def filter(str_input, predicate):
    result = []
    for string in str_input:
        if (predicate(string)):
            result.append(string)
    return result

def display(str_input):
    for str in str_input:
        print(str)

def startsWithA(string):
    if (string[0] == "A"):
        return True
    else:
        return False

strings = ["Hello", "World", "Am", "Hi", "a"]
filtered_result = filter(strings, startsWithA)
display(filtered_result)